ovn-tester/tests/cluster_density.py

```diff
@@ -23,72 +27,57 @@ def __init__(self, config, central_node, worker_nodes, global_cfg):
-            # create 4 legacy pods per iteration.
-            ports = ovn.provision_ports(
-                DENSITY_N_PODS * self.config.n_startup, passive=True
-            )
-            for i in range(self.config.n_startup):
-                ns = Namespace(ovn, f'NS_density_{i}', global_cfg)
-                ns.add_ports(
-                    ports[DENSITY_N_PODS * i : DENSITY_N_PODS * (i + 1)]
-                )
-                ns.create_load_balancer()
-                ovn.provision_lb(ns.load_balancer)
-                ns.provision_vips_to_load_balancers(
-                    [
-                        ports[i * DENSITY_N_PODS : i * DENSITY_N_PODS + 2],
-                        [ports[i * DENSITY_N_PODS + 2]],
-                        [ports[i * DENSITY_N_PODS + 3]],
-                    ]
-                )
+            for index in range(self.config.n_startup):
+                ns = Namespace(ovn, f'NS_density_{index}', global_cfg)
                 all_ns.append(ns)
+                self.run_iteration(ovn, ns, index, global_cfg, passive=True)
 
         with Context(
             ovn,
             'cluster_density',
-            (self.config.n_runs - self.config.n_startup) // self.batch_ratio,
+            self.config.n_runs - self.config.n_startup,
             test=self,
         ) as ctx:
             for i in ctx:
-                ns = Namespace(
-                    ovn, f'NS_density_{self.config.n_startup + i}', global_cfg
-                )
-                ns.create_load_balancer()
-                ovn.provision_lb(ns.load_balancer)
+                index = self.config.n_startup + i
+
+                ns = Namespace(ovn, f'NS_density_{index}', global_cfg)
                 all_ns.append(ns)
 
-                iter_ports = ovn.provision_ports(self.config.batch)
-                for j in range(self.batch_ratio):
-                    # create 6 short lived "build" pods
-                    build_ports = iter_ports[
-                        j * DENSITY_N_TOT_PODS : j * DENSITY_N_TOT_PODS
-                        + DENSITY_N_BUILD_PODS
-                    ]
-                    ns.add_ports(build_ports)
-                    ovn.ping_ports(build_ports)
-                    # create 4 legacy pods
-                    ports = iter_ports[
-                        j * DENSITY_N_TOT_PODS
-                        + DENSITY_N_BUILD_PODS : (j + 1) * DENSITY_N_TOT_PODS
-                    ]
-                    ns.add_ports(ports)
-                    # add VIPs and backends to namespace load-balancer
-                    ns.provision_vips_to_load_balancers(
-                        [ports[0:2], ports[2:3], ports[3:4]]
-                    )
+                # Create DENSITY_N_BUILD_PODS short lived "build" pods.
+                build_ports = ovn.provision_ports(
+                    DENSITY_N_BUILD_PODS, passive=False
+                )
+                ns.add_ports(build_ports)
+                ovn.ping_ports(build_ports)
```

Comment:
Should all that also be part of the iteration?  I'm guessing the port creation/deletion churn might be a valid part of the test.  It will be reflected in the ovn-installed latency at least.  Not sure.
If we'll go that path, we can move everything into the iteration function and have maybe an extra argument to mark if we need to execute a ping test or not.

Reply:
That's a fair point, I'll do it like that, thanks for the review!

Reply:
I updated the code as discussed and I did the same thing for density-heavy (included the ping in the run_iteration function).